metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -264,6 +265,7 @@ class WorksheetProvider(
     if (newDigest != previousDigest) {
       worksheetsDigests.put(path, newDigest)
       val sourceDeps = fetchDependencySources(worksheet.dependencies().asScala)
+      indexWorkspace()
```

Comment:
Good point. I don't actually think the indexing is needed here now if we aren't going to do the workspace symbols, so I'll play around with triggering a compilation here instead to fix that issue.

Reply:
So with playing around with it, indeed, if I trigger a full compile before the evaluation (like 10 lines above this), then opening a worksheet first that includes someone from another Scala file works. However, as you can imagine in a huge codebase, we don't want to do that. Do you think it's fine to instead just to trigger compilation in the build target that holds the worksheet?

Reply:
Should be fine to only compile the build target I think. It will try trigger any other build targets that are needed. That could still take a while, but I don't think there is any other choice here really? :thinking:

Reply:
Alright, so I got this changed around now. I tested it in Metals, and it's actually not that bad. So I think this should work.